yt_dlp/YoutubeDL.py

```diff
@@ -603,10 +609,31 @@ def __init__(self, params=None, auto_init=True):
         except Exception as e:
             self.write_debug(f'Failed to enable VT mode: {e}')
 
-        self._allow_colors = Namespace(**{
-            type_: not self.params.get('no_color') and supports_terminal_sequences(stream)
-            for type_, stream in self._out_files.items_ if type_ != 'console'
-        })
+        color_policy = self.params.get('color') or {}
+        self.params['color'] = {
+            name: (
+                'nocolor' if self.params.get('no_color') else
+                color_policy if isinstance(color_policy, str) else
+                color_policy.get(name) or 'auto')
+            for name in ('screen', 'stdout', 'stderr')
+        }
+        term_allow_color = os.environ.get('TERM', '').lower() != 'dumb'
+
+        def process_color_policy(name, stream):
+            policy = self.params['color'].get(name)
+            if policy == 'never':
+                return False
+            if policy == 'always':
+                return True
+            if policy == 'nocolor':
+                return 'nocolor'
+            return term_allow_color and supports_terminal_sequences(stream)
+
+        self._allow_colors = Namespace(
+            out=process_color_policy('stdout', self._out_files.out),
+            error=process_color_policy('stderr', self._out_files.error),
+            screen=process_color_policy('screen', self._out_files.screen),
```

Comment:
In output rework I swapped `screen` and `out` bc their use seemed to get mixed up. How should we handle this here? Should this stay the way it is and also be applied to output rework or should we try to apply it here as well?